Makefile: change GEOS symlink to a copy

otan opened the pull request:
Instead of symlinking to the correct GEOS directory, force a copy
following the symlinks instead. This dramatically simplifies release
scripts from needing to follow symlinks to add the file to the
archive/Docker container.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50881)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50881#-:-MB9uHqX4Mj5z0JPSdcZ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis)



<!-- Sent from Reviewable.io -->

otan committed:
Makefile: change GEOS symlink to a copy

Instead of symlinking to the correct GEOS directory, force a copy
following the symlinks instead. This dramatically simplifies release
scripts from needing to follow symlinks to add the file to the
archive/Docker container.

Release note: None

otan commented:
bors r=petermattis
